File: quantum-vulnerable/BrokenAlgorithms.java
package quantum.vulnerable;

import java.security.KeyPair;
import java.security.KeyPairGenerator;
import java.security.Signature;
import java.security.spec.ECGenParameterSpec;
import javax.crypto.Cipher;
import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import javax.crypto.spec.IvParameterSpec;

/** Quantum-vulnerable / broken algorithms: RSA (Shor), ECDSA (Shor), 3DES (weak). */
public final class BrokenAlgorithms {

    public static byte[] rsaOaep(byte[] data) throws Exception {
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
        kpg.initialize(2048);
        KeyPair kp = kpg.generateKeyPair();
        Cipher c = Cipher.getInstance("RSA/ECB/OAEPWithSHA-256AndMGF1Padding");
        c.init(Cipher.ENCRYPT_MODE, kp.getPublic());
        return c.doFinal(data);
    }

    public static byte[] ecdsaSign(byte[] message) throws Exception {
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("EC");
        kpg.initialize(new ECGenParameterSpec("secp256r1"));
        KeyPair kp = kpg.generateKeyPair();
        Signature s = Signature.getInstance("SHA256withECDSA");
        s.initSign(kp.getPrivate());
        s.update(message);
        return s.sign();
    }

    public static byte[] tripleDes(byte[] data) throws Exception {
        SecretKey key = KeyGenerator.getInstance("DESede").generateKey();   // 3DES, deprecated
        Cipher c = Cipher.getInstance("DESede/CBC/PKCS5Padding");
        c.init(Cipher.ENCRYPT_MODE, key, new IvParameterSpec(new byte[8]));
        return c.doFinal(data);
    }

    public static void main(String[] args) throws Exception {
        System.out.println("RSA-OAEP ciphertext: " + rsaOaep("secret".getBytes()).length);
        System.out.println("ECDSA signature: " + ecdsaSign("message".getBytes()).length);
        System.out.println("3DES ciphertext: " + tripleDes("data".getBytes()).length);
    }
}

File: quantum-vulnerable/broken_algorithms.rs
// Quantum-vulnerable / broken algorithms: RSA (Shor), ECDSA (Shor), 3DES (weak).
// crates: rsa, sha2, rand, p256, ecdsa, des
use des::cipher::generic_array::GenericArray;
use des::cipher::{BlockEncrypt, KeyInit};
use des::TdesEde3;
use p256::ecdsa::signature::Signer;
use p256::ecdsa::{Signature, SigningKey};
use rand::rngs::OsRng;
use rsa::{Oaep, RsaPrivateKey};
use sha2::Sha256;

pub fn rsa_oaep(data: &[u8]) -> Vec<u8> {
    let mut rng = OsRng;
    let private_key = RsaPrivateKey::new(&mut rng, 2048).expect("rsa keygen");
    private_key
        .to_public_key()
        .encrypt(&mut rng, Oaep::new::<Sha256>(), data)
        .expect("rsa encrypt")
}

pub fn ecdsa_sign(message: &[u8]) -> Signature {
    let signing_key = SigningKey::random(&mut OsRng);
    signing_key.sign(message)
}

pub fn triple_des(key: &[u8; 24], block: &mut [u8; 8]) {
    let cipher = TdesEde3::new(GenericArray::from_slice(key)); // 3DES, deprecated
    cipher.encrypt_block(GenericArray::from_mut_slice(block));
}
